### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: DEV-GUIDE.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Simperby Development Guide
 
+## Checklist
+
+- run `cargo fmt`
+- run `cargo clippy --all --all-targets --all-features`
+- run `cargo test --all --all-targets --all-features`
```
Comment (TomTaehoonKim) [inline DEV-GUIDE.md:7]: why not the same as the commands in the ci?

File: DEV-GUIDE.md [hunk 2]
```diff
@@ -38,16 +44,16 @@ We have to verify 3 things here:
...
-- Use **rebase only** (No merge or squash).
+- Use **rebase only**.
 - **NO MERGE COMMITS**.
 - Push your works on your own forked repo, and make a **PR across forks**.
 - Use the same title (head) and content (body) for a single-commit PR.
 - When the PR resolves some issues, put `fixes #123` in the content of the PR, not in the commit messages.
 - Conversation **must be resolved by the commentor**, not the PR author.
-- Do not take the Github notifications as the communication channel; use Discord.
-- To resolve reviews, always put additional commits representing the changes, not amend the existing commits and force push it.
+- Do not take the Github notifications as the communication channel; use Discord and directly mention the relevant people.
```
Comment (TomTaehoonKim) [inline DEV-GUIDE.md:53]: I was curious with this from the past actually.

> Do not take the Github notifications as the communication channel; use Discord
Does this sentence mean not to use Github comments to do discussions? but to use Discord?

Comment (junha1) [reply DEV-GUIDE.md]: Clarified the sentence.

File: DEV-GUIDE.md (hunk 1 above)
Comment (junha1) [reply DEV-GUIDE.md:7]: `-D warnings` is for emitting an error if failed, to notify the CI.
For `cargo test`, I updated the CI.

**Review by TomTaehoonKim — APPROVED** [review]

**Pull request merged**